CMakeLists.txt

```diff
@@ -0,0 +1,934 @@
+	git-request-pull git-submodule git-web--browse
+	git-mergetool--lib git-parse-remote git-rebase--preserve-merges
+	git-sh-setup git-sh-i18n git-instaweb)
+
+foreach(scr ${git_shell_scripts})
+	file(STRINGS ${CMAKE_SOURCE_DIR}/${scr}.sh content NEWLINE_CONSUME)
+	string(REPLACE "@SHELL_PATH@" "/bin/sh" content "${content}")
+	string(REPLACE "@@DIFF@@" "diff" content "${content}")
+	string(REPLACE "@LOCALEDIR@" "${LOCALEDIR}" content "${content}")
+	string(REPLACE "@GITWEBDIR@" "${GITWEBDIR}" content "${content}")
+	string(REPLACE "@@NO_CURL@@" "" content "${content}")
+	string(REPLACE "@@USE_GETTEXT_SCHEME@@" "" content "${content}")
+	string(REPLACE "# @@BROKEN_PATH_FIX@@" "" content "${content}")
+	string(REPLACE "@@PERL@@" "/usr/bin/perl" content "${content}")
+	string(REPLACE "@@SANE_TEXT_GREP@@" "-a" content "${content}")
+	string(REPLACE "@@PAGER_ENV@@" "LESS=FRX LV=-c" content "${content}")
+	file(WRITE ${CMAKE_BINARY_DIR}/${scr} ${content})
+endforeach()
+
+set(git_perl_scripts
+git-add--interactive git-archimport git-cvsexportcommit 
+git-cvsimport git-cvsserver git-send-email git-svn) 
+
+
+file(STRINGS ${CMAKE_SOURCE_DIR}/perl/header_templates/fixed_prefix.template.pl perl_header )
+string(REPLACE "@@PATHSEP@@" ":" perl_header "${perl_header}")
+string(REPLACE "@@INSTLIBDIR@@" "${INSTLIBDIR}" perl_header "${perl_header}")
+#message(${perl_header})
+
+foreach(scr ${git_perl_scripts})
+	file(STRINGS ${CMAKE_SOURCE_DIR}/${scr}.perl content NEWLINE_CONSUME)
+	string(REPLACE "#!/usr/bin/perl" "#!/usr/bin/perl\n${perl_header}\n" content "${content}")
+	string(REPLACE "@@GIT_VERSION@@" "${PROJECT_VERSION}" content "${content}")
+	file(WRITE ${CMAKE_BINARY_DIR}/${scr} ${content})
+endforeach()
+
+
+file(STRINGS ${CMAKE_SOURCE_DIR}/git-p4.py content NEWLINE_CONSUME)
+string(REPLACE "#!/usr/bin/env python" "#!/usr/bin/python" content "${content}")
+file(WRITE ${CMAKE_BINARY_DIR}/git-p4 ${content})
+
+
+#perl modules
+file(GLOB_RECURSE perl_modules "${CMAKE_SOURCE_DIR}/perl/*.pm")
+
+foreach(pm ${perl_modules})
+	string(REPLACE "${CMAKE_SOURCE_DIR}/perl/" "" file_path ${pm})
+	file(STRINGS ${pm} content NEWLINE_CONSUME)
+	string(REPLACE "@@LOCALEDIR@@" "${LOCALEDIR}" content "${content}")
+	string(REPLACE "@@NO_PERL_CPAN_FALLBACKS@@" "" content "${content}")
+	file(WRITE ${CMAKE_BINARY_DIR}/perl/build/lib/${file_path} ${content})
+#test-lib.sh requires perl/build/lib to be the build directory of perl modules
+endforeach()
+
+
+#templates
+file(MAKE_DIRECTORY ${CMAKE_BINARY_DIR}/templates/blt/branches) #create branches
+set(hooks_templates
+applypatch-msg.sample      pre-applypatch.sample      pre-push.sample
+commit-msg.sample          pre-commit.sample          pre-rebase.sample
+fsmonitor-watchman.sample  pre-merge-commit.sample    pre-receive.sample
+post-update.sample         prepare-commit-msg.sample  update.sample)
+
+#templates have @.*@ replacement so use configure_file instead
+set(SHELL_PATH /bin/sh)
+set(PERL_PATH /usr/bin/perl)
```

Comment:
We should probably define those earlier, and use them when copy-editing the `*.sh` and `*.perl` scripts already, instead of hard-coding, say, `/bin/sh` multiple times.

Reply:
I can make that change.
I have a question, /usr/bin/perl means that you have perl interpreter in that location resolved by bash. The perl that ships with the Git-for-windows installer does not allow you to install any modules. So wouldn't it be better if I were to install strawberry perl in say C:\perl and point PERL_PATH to that. Will this work?

Reply:
I don't think that Strawberry Perl will work in general, due to Windows path problems (IIRC there are quite a couple places where Git's Perl scripts expect absolute paths to start with `/`, and path lists like `PATH` to be separated by `:`, not `;`), and due to lack of the svn-perl modules.